### ISSUE TITLE
Refactor to use normal loop and Stud.stoppable_sleep

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-input-example#6]
As it is now, the run method is quite confusing since it uses the Stop.stop!, and puts the current thread on an ivar, etc etc:

``` ruby
  def run(queue)
    # This plugin uses Stud.interval. Because we want it to be able to stop
    # cleanly, we need to keep access to the current thread.  It is referenced
    # in the stop method below.
    #
    # Other plugins may require similar access to the current thread.
    @thread = Thread.current

    Stud.interval(@interval) do
      event = LogStash::Event.new("message" => @message, "host" => @host)
      decorate(event)
      queue << event
    end # loop
  end # def run

  def stop
    Stud.stop!(@thread)
  end
```

I think it would be better and more understandable to say instead

``` ruby
  def run(queue)
    loop do
      event = LogStash::Event.new("message" => @message, "host" => @host)
      decorate(event)
      queue << event
      Stud.stoppable_sleep(@interval) { stop? }
      break if stop?
    end # loop
  end # def run
```

Ideas?

**Comment by ph** [comment]
I agree the example is now more confusing for new plugin author and unless they know the internals of `stud` it will be hard to understand.

I am okay with your changes @jsvd, but maybe remove the break and use this construct?

``` ruby
while !stop?
#...
end
```

**Comment by jsvd** [comment]
fixed by #7
